src/operations/blas3/gemm_no_local.hpp

```diff
@@ -142,53 +127,61 @@ class Gemm<input_t, output_t, DoubleBuffer, NbcA, NbcB, ClSize, tile_type,
    *research.
    *
    */
-  static SYCL_BLAS_INLINE index_t get_num_workgroup_cluster(
-      index_t m, index_t n, index_t compute_units) noexcept {
+  SYCL_BLAS_INLINE index_t
+  get_num_workgroup_cluster(index_t compute_units) const noexcept {
     constexpr index_t num_gemm_per_compute_units = 4;
     return ((num_gemm_per_compute_units * compute_units - 1) /
-                get_workgroup_cluster(m, n) +
+                get_workgroup_cluster() +
             1);
   }
 
-  static SYCL_BLAS_INLINE cl::sycl::nd_range<1> get_nd_range(
-      index_t m, index_t n, index_t compute_units) noexcept {
-    const cl::sycl::range<1> nwg(
-        get_workgroup_cluster(m, n) *
-        get_num_workgroup_cluster(m, n, compute_units));
+  SYCL_BLAS_INLINE cl::sycl::nd_range<1> get_nd_range(
+      index_t compute_units) const noexcept {
+    const cl::sycl::range<1> nwg(get_workgroup_cluster() *
+                                 get_num_workgroup_cluster(compute_units));
     const cl::sycl::range<1> wgs(wg_rows * wg_cols);
 
     return cl::sycl::nd_range<1>(nwg * wgs, wgs);
   }
 
-  SYCL_BLAS_INLINE index_t get_size() const { return m_ * n_; }
+  SYCL_BLAS_INLINE index_t get_size() const {
+    return a_.get_size_row() * b_.get_size_col();
+  }
 
   SYCL_BLAS_INLINE bool valid_thread(cl::sycl::nd_item<1> ndItem) const {
     return true;
   }
 
   SYCL_BLAS_INLINE void eval(cl::sycl::nd_item<1> id) noexcept {
```

Comment:
here as well change it to either cl::sycl::nd_item<1>& or const cl::sycl::nd_item<1>&